### PR TITLE
Toolchain files simplification and build fixes

### PR DESCRIPTION
**Pull request by Flamefire**
Factor out common parts of toolchains.

Fix language handling (gettext)

### CODE REVIEW

File: cmake/toolchains/c.linux.i686.cmake [hunk 1]
```diff
@@ -1,22 +1,6 @@
...
 
-# search for programs in the build host directories
-# for libraries and headers in the target directories
-SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+include(${CMAKE_CURRENT_LIST_DIR}/c.linux.common.cmake)
 
-SET(Boost_COMPILER "-gcc44")
+set(Boost_COMPILER "-gcc44")
```
Comment (Flow86) [inline cmake/toolchains/c.linux.i686.cmake]: das können wir entfernen

File: cmake/toolchains/c.linux.x86_64.cmake [hunk 2]
```diff
@@ -1,22 +1,6 @@
...
 
-# search for programs in the build host directories
-# for libraries and headers in the target directories
-SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+include(${CMAKE_CURRENT_LIST_DIR}/c.linux.common.cmake)
 
-SET(Boost_COMPILER "-gcc44")
+set(Boost_COMPILER "-gcc44")
```
Comment (Flow86) [inline cmake/toolchains/c.linux.x86_64.cmake]: hier ebenso

**Comment by Flamefire** [comment]
@Flow86 Added some more fixes. Also had to modify master as I accidentally removed referenced commits from libsiedler2 :/ 

Could we keep development in branches/PRs until all CI stuff passes and squash successive fixup commits before merging?

File: cmake/optimizations.cmake [hunk 3]
```diff
@@ -22,10 +22,10 @@ else()
...
         endif()
     else()
-        set_if_unset(RTTR_OPTIMZATION_VECTOR_EXT_DEFAULT SSE)
+        set_if_unset(RTTR_OPTIMZATION_VECTOR_EXT_DEFAULT SSE2)
         set(RTTR_OPTIMZATION_VECTOR_EXT ${RTTR_OPTIMZATION_VECTOR_EXT_DEFAULT} CACHE STRING "Vector extension to use")
         # Set allowed options
-        set_property(CACHE RTTR_OPTIMZATION_VECTOR_EXT PROPERTY STRINGS OFF SSE SSE2 SSE3 SSE4 AVX AVX2)
+        set_property(CACHE RTTR_OPTIMZATION_VECTOR_EXT PROPERTY STRINGS OFF SSE2 SSE3 SSE4 AVX AVX2)
         if(RTTR_OPTIMZATION_VECTOR_EXT)
             if(MINGW AND RTTR_OPTIMZATION_VECTOR_EXT STREQUAL "SSE")
```
Comment (Flow86) [inline cmake/optimizations.cmake]: we could remove this if then, couldnt we?

Comment (Flow86) [reply cmake/optimizations.cmake]: oh and optim**i**zation is spelled wrong

**Comment by Flamefire** [comment]
@Flow86 Could I also get merge rights to the protected branch(es)?

**Comment by Flow86** [comment]
you should have, you only need a review approval ;-)

**Comment by Flow86** [comment]
(oh and the 2 required checks, travis-ci and jenkins have to be run through)

**Comment by Flamefire** [comment]
Still can't merge although both CIs passed and you approved.

**Comment by Flow86** [comment]
now it should work I think

**Pull request merged**